database/rrdset.c

```diff
@@ -1463,17 +1463,28 @@ void rrdset_done(RRDSET *st) {
                     uint64_t max = (uint64_t)rd->collected_value_max;
                     uint64_t cap = 0;
 
-                         if(max > 0x00000000FFFFFFFFULL) cap = 0xFFFFFFFFFFFFFFFFULL;
-                    else if(max > 0x000000000000FFFFULL) cap = 0x00000000FFFFFFFFULL;
-                    else if(max > 0x00000000000000FFULL) cap = 0x000000000000FFFFULL;
-                    else                                 cap = 0x00000000000000FFULL;
+                    if (max > 0x00000000FFFFFFFFULL)
+                        cap = 0xFFFFFFFFFFFFFFFFULL;
+                    else
+                        cap = 0x00000000FFFFFFFFULL;
```

Comment:
While I agree that 8-bit, 16-bit, and arbitrary precision counters should just be ignored by this code, I do think we should handle cases of signed 32 and 64 bit counters (wrap points would be 0x000000007FFFFFFFULL and 0xFFFFFFFFFFFFFFFULL respectively). There is, unfortunately, some software (and hardware) out there that makes the poor choice of using signed types for counters, and the likelihood of either transition being a false positive is pretty low once we factor in the below logic.